processing/api.py

```diff
@@ -122,3 +127,65 @@ def add_track_tag(self, recording, track_id, prediction, data=""):
         if r.status_code == 200:
             return r.json()["trackTagId"]
         raise IOError(r.text)
+
+    def download_file(self, token, filename):
+        r = requests.get(
+            urljoin(self.api_url, "/api/v1/signedUrl"),
+            params={"jwt": token},
+            stream=True,
+        )
+        r.raise_for_status()
+        return iter_to_file(filename, r.iter_content(chunk_size=4096))
+
+    def upload_file(self, token, filename):
+        r = requests.get(
+            urljoin(self.api_url, "/signedUrl"),
+            params={"jwt": token},
+            stream=True,
+        )
+        r.raise_for_status()
+        return iter_to_file(filename, r.iter_content(chunk_size=4096))
+
+    def upload_file(self, filename):
+        url = self.file_url + "/processed"
+        data = {"fileHash": sha_hash(filename)}
```

Comment:
Could be wrong, but looks like you're hashing the filename here rather than the file contents?